utils/config.go

```diff
@@ -760,6 +760,109 @@ func GenerateClientConfig(c *model.Config, diagnosticId string, license *model.L
 	return props
 }
 
+func GenerateLimitedClientConfig(c *model.Config, diagnosticId string, license *model.License) map[string]string {
```

Comment:
I'm worried about the longer-term duplication here over what's in `GenerateClientConfig`, in terms of maintenance and "forgetting" to update one or the other.

Since the limited config is a subset of the full config, could we have `GenerateClientConfig` call `GenerateLimitedClientConfig` and then add in the missing fields? Then it becomes a conscious decision with each new configuration bit whether to expose it by default or not, and we avoid duplication.

Reply:
I like that idea much better. I was trying to figure out a better way to do it without all the duplication, but couldn't come up with anything myself